packages/idyll-document/src/index.js

```diff
@@ -99,6 +99,12 @@ class Wrapper extends React.PureComponent {
     if (props.hasHook || exps.some(v => v.includes('refs.'))) {
       updateRefsCallbacks.push(this.onUpdateRefs);
     }
+
+    this.state = { hasError: false, errorMessage: '' };
+  }
+
+  componentDidCatch(error, info) {
+    this.setState({ hasError: true, errorMessage: error.message });
```

Comment:
Should we store the whole error object instead? Also, and more importantly, I was getting a message about cross origin errors that you didn't seem to be. I'm not sure if this is due to a new React build or what. Any ideas?

Reply:
Storing the whole object seems like a reasonable thing to do. This is the only thing I could find https://facebook.github.io/react/docs/cross-origin-errors.html#webpack, I guess I was running a production build and didn't see this, hmmm

Reply:
Agreed on the whole object.